dwave/optimization/include/dwave-optimization/nodes/collections.hpp

```diff
@@ -221,12 +221,18 @@ class DisjointListsNode : public DecisionNode {
 
     void rotate_in_list(State& state, ssize_t list_index, ssize_t dest_idx, ssize_t src_idx) const;
 
+    void rotate_sublist_right(State& state, ssize_t list_index, ssize_t start, ssize_t stop,
+                              ssize_t num_rotations) const;
+
     void swap_in_list(State& state, ssize_t disjoint_list, ssize_t element_i,
                       ssize_t element_j) const;
 
     void pop_to_list(State& state, ssize_t from_disjoint_list, ssize_t element_i,
                      ssize_t to_disjoint_list, ssize_t element_j) const;
 
+    void set_full_list_state(State& state, ssize_t list_index,
```

Comment:
Feels like `set_state(...)` covers this and is a bit more succinct?

Also, we should accept a `span` rather than a `vector`.